plugins/notifications-node/src/extensions.ts

```diff
@@ -100,13 +100,29 @@ export interface NotificationProcessor {
   getNotificationFilters?(): NotificationProcessorFilters;
 }
 
+/**
+ * NotificationRecipientResolver is a function that resolves the individual users to receive the notification
+ * based on the entity reference(s) and the excluded entity reference(s).
+ *
+ * The function should return a list of user entity references that should receive the notification.
+ *
+ * @public
+ */
+export type NotificationRecipientResolver = (
+  entityRef: string | string[] | null,
+  excludeEntityRefs: string | string[],
```

Comment:
This should probably be a single options argument (or perhaps having the entityRef as one argument and then an options object as the second, for extensibility. That's a pattern we use a lot. Also should `excludeEntityRefs` be optional? And should the return type also be an object with like `recipients: string[]`?

Reply:
The extension is following the same way `notificationsService` is using these. Changing those would of course be a breaking change. That's the reason why these are as they are :D

Reply:
Actually we could just pass the entityRefs and handle exclusion in the notifications backend. But you are right, I will modify this a bit 👍